### PR TITLE
Fix metrics #217

### PR DESCRIPTION
**Pull request by rocrisp**
Fixed a few issues.
1. Place upgrade counter in the correct location.
2. Handle empty string

### CODE REVIEW

File: internal/sub_reconcilers/reconcile_deployment_updates.go [hunk 1]
```diff
@@ -239,10 +243,17 @@ func getPodNamesandStatuses(pods []corev1.Pod) []string {
...
-				return 1
+		pendingState := string(pod.Status.Phase)
+		if pendingState == "Pending" {
+			if len(pod.Status.ContainerStatuses) > 0 {
+				if len(pod.Status.ContainerStatuses[0].State.Waiting.Reason) > 0 {
+					errorImagePull := string(pod.Status.ContainerStatuses[0].State.Waiting.Reason)
+					imagePullBackOff := string(pod.Status.ContainerStatuses[0].State.Waiting.Reason)
+					if errorImagePull == "ErrImagePull" ||
+						imagePullBackOff == "ImagePullBackOff" {
+						return 1
```
Comment (skattoju) [inline internal/sub_reconcilers/reconcile_deployment_updates.go:254]: does this have to be a float64 ?

File: internal/sub_reconcilers/reconcile_deployment_updates.go [hunk 2]
```diff
@@ -239,10 +243,17 @@ func getPodNamesandStatuses(pods []corev1.Pod) []string {
...
 	// return 0 if not found, otherwise return 1
 	for _, pod := range pods {
-		if string(pod.Status.Phase) == "Pending" {
-			if string(pod.Status.ContainerStatuses[0].State.Waiting.Reason) == "ErrImagePull" ||
-				string(pod.Status.ContainerStatuses[0].State.Waiting.Reason) == "ImagePullBackOff" {
-				return 1
+		pendingState := string(pod.Status.Phase)
+		if pendingState == "Pending" {
+			if len(pod.Status.ContainerStatuses) > 0 {
+				if len(pod.Status.ContainerStatuses[0].State.Waiting.Reason) > 0 {
```
Comment (skattoju) [inline internal/sub_reconcilers/reconcile_deployment_updates.go:249]: could combine these into a single if with &&s right ?

File: internal/sub_reconcilers/reconcile_deployment_updates.go (hunk 1 above)
Comment (rocrisp) [reply internal/sub_reconcilers/reconcile_deployment_updates.go:254]: I think so, prometheus.NewGauge returns float64

File: internal/sub_reconcilers/reconcile_deployment_updates.go (hunk 2 above)
Comment (rocrisp) [reply internal/sub_reconcilers/reconcile_deployment_updates.go:249]: good idea

**Comment by rocrisp** [comment]
@skattoju I yolo merged this one by mistake

**Review by skattoju — APPROVED** [review]
/lgtm

**Pull request merged**